### PR TITLE
Merge toys again

### LINKED ISSUE #7: Merge with toy gridworlds
To avoid code duplication perhaps we could merge this repo with the toy gridworlds one (whichever way is more convenient). This would be especially nice for having a consistent interface for corrupted gridworlds and would also make solving #6 easier.

### PR DESCRIPTION
**Pull request by david-lindner**
Builds on #10 and #8 and closes #7

Addresses my comments regarding the render method in #10 and changes the font as discussed in #8 

Also adds some missing dependencies and very basic unittests to check that registering the environments works.

### CODE REVIEW

File: safe_grid_gym/envs/common/base_gridworld.py [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+
+                plt.axis("off")
+                plt.imshow(image)
+                plt.pause(0.1)
+                plt.clf()
+            elif mode == "rgb_array":
+                return np.array(image)
+            else:
+                # unknown mode
+                raise NotImplementedError
```
Comment (jvmncs) [inline safe_grid_gym/envs/common/base_gridworld.py]: might want to add an error message here via `NotImplementedError("Mode unsupported blah blah")`

File: safe_grid_gym/tests/test_toy_gridworlds.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+            self.assertIsInstance(env.render(mode="rgb_array"), np.ndarray)
+
+            matplotlib.use("Agg")
+            env.render(mode="human")
+
+            # "text" is not a valid mode
+            with self.assertRaises(NotImplementedError):
+                env.render(mode="text")
+
+    def testWithRandomMoves(self):
```
Comment (jvmncs) [inline safe_grid_gym/tests/test_toy_gridworlds.py:34]: unittests with randomness should be seeded for reproducibility

File: safe_grid_gym/envs/common/base_gridworld.py (hunk 1 above)
Comment (david-lindner) [reply safe_grid_gym/envs/common/base_gridworld.py]: fixed

File: safe_grid_gym/tests/test_toy_gridworlds.py (hunk 2 above)
Comment (david-lindner) [reply safe_grid_gym/tests/test_toy_gridworlds.py:34]: fixed

File: setup.py [hunk 3]
```diff
@@ -60,7 +60,7 @@
         "matplotlib",
     ],
     dependency_links=[
-        "https://github.com/jvmancuso/gym/tarball/master#egg=gym-0.10.10",
+        "git+https://github.com/openai/gym.git#egg=gym-0.10.9",
```
Comment (jvmncs) [inline setup.py:63]: have you tested the toy gridworlds in a newly installed virtualenv with this dependency link? here's the bug I was getting before with a similar link:
```python
>>> import gym
>>> import safe_grid_gym
>>> env = gym.make("ToyGridworldCorners-v0")
/Users/jasonmancuso/anaconda/envs/safe-grid-agents/lib/python3.5/site-packages/gym-0.10.9-py3.5.egg/gym/envs/registration.py:14: PkgResourcesDeprecationWarning: Parameters to load are deprecated.  Call .resolve and .require separately.
  result = entry_point.load(False)```

Comment (david-lindner) [reply setup.py:63]: Yes this seems to work for me.

I set up an environment like this:

```
conda create -n "test-gym" python=3.6
conda activate test-gym
python setup.py develop
```

then I see during the installation of dependencies

```
...
Searching for gym
Doing git clone from https://github.com/openai/gym.git to /tmp/easy_install-mg83u4aq/gym.git
...
```

For some reason I also get an

```
Writing /tmp/easy_install-mg83u4aq/gym.git/setup.cfg
Running setup.py -q bdist_egg --dist-dir /tmp/easy_install-mg83u4aq/gym.git/egg-dist-tmp-0m1ylint
  File "build/bdist.linux-x86_64/egg/gym/utils/reraise_impl_py2.py", line 2
    raise e.__class__, e, traceback
                     ^
SyntaxError: invalid syntax

creating /home/david/anaconda3/envs/test-gym/lib/python3.6/site-packages/gym-0.10.9-py3.6.egg
Extracting gym-0.10.9-py3.6.egg to /home/david/anaconda3/envs/test-gym/lib/python3.6/site-packages
  File "/home/david/anaconda3/envs/test-gym/lib/python3.6/site-packages/gym-0.10.9-py3.6.egg/gym/utils/reraise_impl_py2.py", line 2
    raise e.__class__, e, traceback
                     ^
SyntaxError: invalid syntax
```

but gym seems to build fine apart from that :sweat_smile:

Then I start a python interpreter

```
> which python
/home/david/anaconda3/envs/test-gym/bin/python
> python
```

and your code runs without the error:

```
Python 3.6.8 |Anaconda, Inc.| (default, Dec 30 2018, 01:22:34) 
[GCC 7.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import gym
>>> import safe_grid_gym
>>> env = gym.make("ToyGridworldCorners-v0")
```

**Pull request merged**